src/main/java/org/opentripplanner/graph_builder/linking/SimpleStreetSplitter.java

```diff
@@ -517,7 +517,12 @@ public Vertex getClosestVertex(GenericLocation location, RoutingRequest options,
         if (options != null) {
             TraverseModeSet modes = options.modes;
             if (modes.getCar())
-                nonTransitMode = TraverseMode.CAR;
+                // for park and ride we will start in car mode and walk to the end vertex
+                if(endVertex && (options.parkAndRide || options.kissAndRide)){
```

Comment:
nit: please correct the formatting on this line (and the else) to match the rest of OTP.